sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -191,9 +193,25 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
       if pipeline_conf.timeout:
         pipelinerun['spec']['timeout'] = '%ds' % pipeline_conf.timeout
 
-
       workflow = workflow + [pipelinerun]
 
+    # Use regex to replace all the Argo variables to Tekton variables.
+    # For variables that are unique to Argo, we prompt a warning to warn users on the non-supported variables.
+    # Since Argo variables can be used in anywhere in the yaml, we need to dump and then parse the whole yaml
+    # using regular expression.
+    workflow_dump = json.dumps(workflow)
+    tekton_var_regex_rules = [
+        {'argo_rule': '{{inputs.parameters.([^ \t\n.:,;{}]+)}}', 'tekton_rule': '$(inputs.params.\g<1>)'},
+        {'argo_rule': '{{outputs.parameters.([^ \t\n.:,;{}]+).path}}', 'tekton_rule': '$(results.\g<1>.path)'}
+    ]
+    for regex_rule in tekton_var_regex_rules:
+      workflow_dump = re.sub(regex_rule['argo_rule'], regex_rule['tekton_rule'], workflow_dump)
+
+    unsupported_vars = re.findall(r"{{[^ \t\n:,;{}]+}}", workflow_dump)
+    if unsupported_vars:
+      logging.warning('These Argo variables are not supported in Tekton Pipeline: %s' % ", ".join(str(v) for v in set(unsupported_vars)))
```

Comment:
I am assuming that if one of the Argo specific variables is not supported, then the resulting Tekton pipeline does not behave in the way the author of the Python DSL script intended it to behave. In which case I think should be a little stronger here and `raise TypeError` or `raise ValueError` or even `raise AttributeError`. No?

Reply:
Let me see is there other places that rely on this. If not then I can change it to raise and exit.